### DISCUSSION TITLE
Cannot find cloud discovery option

### DISCUSSION THREAD

**Discussion by CometStrike** [krahabb/meross_lan#316] [Q&A] [answered]
Hi, I've recently gotten into Home Assistant and this is my first HACS integration. My MSS310 devices are showing in the discovered section in HA, however, it directly asks me for a device key: 
![grafik](https://github.com/krahabb/meross_lan/assets/144691735/525b2ccb-21f3-4ed7-b9f4-dc21b422a6a6)
The Readme says that the easiest integration is via Meross cloud access, which I had seen in a Youtube video to look something like this:
![grafik](https://github.com/krahabb/meross_lan/assets/144691735/f1aa3490-a8eb-4181-9ede-c6140fe472f7)
Has something changed in configuring the devices since the video has been released (january 2023)? How/where do I add my cloud access data in order to proceed? Or if the device key is needed: where do I find that in the Meross app?

I am using V4.3 of the meross_LAN integration with Home Assistant 2023.9.1

**Comment by CometStrike** [comment, marked as answer]
Found it in the integration under "Setup another instance of Meross LAN"

**Reply by buildintechie** [reply]
@CometStrike, can you provide a little more information on where you found "Setup another instance of Meross LAN"?  I'm having the same problem.  Thanks.

**Reply by krahabb** [reply]
The popup shown in this thread is a bit 'legacy' so it is not available anymore. With current HA/meross_lan UI it works like this:
- if you've just installed meross_lan from scratch you have to `ADD INTEGRATION` from the main HA Integrations page and search for meross_lan through the list of available integrations
- If you've already setup any configuration in meross_lan, just enter the 'Meross LAN' section from the main Integrations page and, at the bottom of the right panel, hit `ADD HUB`

In both cases you'll be prompt by a menu where you're asked for either 'Register a Meross cloud profile' or 'Manually configure a device'
- If you choose to register the cloud profile, it will ask your account info and when set up it will download your account devices and infos and start discovering them (automatically) - provided you've set the option to allow the 'MQTT publish' (this is a kind of protection if you just want to use the cloud brokers in 'listen mode' so you don't risk to build too much traffic against these brokers)
- If you instead want to manually add your device you have to choose the other option and provide device address and key. Again, when you configure a device this way, if you don't know the device key or if it's wrong, the UI will then ask you if you want to setup a Meross cloud profile to retrieve the key for the device (this way you'll configure the Meross account exactly as if you hitted the first option in the main menu)

**Reply by buildintechie** [reply]
@krahabb, I’ve tried to set it up with the cloud profile, however when I enter my credentials the system won’t login…and I’ve reset my password several times to make sure it’s right.  Then, I got this email about 30 minutes after trying a number of time:
![IMG_4216](https://github.com/krahabb/meross_lan/assets/155104130/b2bcd0e1-97ec-44dc-997e-c140670b838d)
![IMG_4217](https://github.com/krahabb/meross_lan/assets/155104130/c3f212da-a5d5-4e6d-b728-dc493ef9efcc)

**Reply by krahabb** [reply]
Yes, 'we have a problem' right now ( see #372).
meross_lan still uses an old api which, from what I've understood, got discontinued this january.

Strangely enough, login with my credentials still work with the old api so I couldnt spot the issue until other users mentioned it

**Comment by krahabb** [comment]
Hello @CometStrike,
Yeah, this is confusing a bit since the discovery process (which works by sniffing the DHCP traffic) is still a bit naive. I should go for an improvement there asking the user if he wants to setup the Meross cloud account directly from the discovery mask
